staging/src/k8s.io/controller-manager/pkg/leadermigration/migrator.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leadermigration
+
+import (
+	internal "k8s.io/controller-manager/config"
+)
+
+type LeaderMigrator struct {
+	config              *internal.LeaderMigrationConfiguration
+	migratedControllers map[string]bool
+	// component indicates the name of the control-plane component that uses leader migration,
+	//  which should be a controller manager, i.e. kube-controller-manager or cloud-controller-manager
+	component string
+}
+
+type FilterFunc func(controllerName string) bool
+
+func NewLeaderMigrator(config *internal.LeaderMigrationConfiguration, component string) *LeaderMigrator {
```

Comment:
Go doc explaining the parameters here and what they do would be helpful. I was able to figure it out from reading the code, but when I first just looked at the signatures of the functions it wasn't clear to me what `component` is how how it relates to what the `FilterFunc` matches.

Reply:
Thanks. Updated the comments.